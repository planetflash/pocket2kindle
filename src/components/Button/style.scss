.btn {
	display: inline-block;
	padding: 1em 2em;
	background: blue;
	color: white;
	cursor: pointer;
}
