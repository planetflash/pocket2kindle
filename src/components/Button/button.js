import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

import { mergePropsWithAvailableProps } from "utils";
import events from "utils/events";
import Shake from "transitions/Shake";

const availableProps = {
  ...events.mouse,
  ...events.touch
};

import "./style.scss";

class Button extends React.Component {
  render() {
    const { text, loading, error } = this.props;

    const classes = classNames({
      btn: true,
      loading
    });

    return (
      <Shake in={error}>
        <a
          styleName={classes}
          {...mergePropsWithAvailableProps(availableProps, this.props)}
        >
          <span>{text}</span>
        </a>
      </Shake>
    );
  }
}

Button.propTypes = {
  text: PropTypes.string,
  loading: PropTypes.bool,
  error: PropTypes.bool
};

export default Button;
