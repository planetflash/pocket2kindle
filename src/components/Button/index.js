import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

import { mergePropsWithAvailableProps } from "../../utils";
import events from "../events";

const availableProps = {
  ...events.mouse,
  ...events.touch
};

import "./style.scss";

class Button extends React.Component {
  render() {
    const classes = classNames({
      btn: true,
      loading: this.props.loading
    });

		const testChange = 'change 2'

    return (
      <a
        className={classes}
        {...mergePropsWithAvailableProps(availableProps, this.props)}
      >
        {this.props.children}
      </a>
    );
  }
}

Button.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
  onClick: PropTypes.func,
  loading: PropTypes.bool
};

export default Button;
