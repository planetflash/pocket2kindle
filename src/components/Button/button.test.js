import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import Button from "./";

describe("Button component", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<Button />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("shows the the correct text", () => {
    const testMessage = "test message";
    const wrapper = shallow(<Button text={testMessage} />);
    const text = wrapper.find(".text").text();

    expect(text).toBe(testMessage);
  });
});
