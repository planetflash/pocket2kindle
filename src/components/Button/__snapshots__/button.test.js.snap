// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches the snapshot 1`] = `
<div>
  <a
    className="src-components-Button-___style__btn___2qMxX"
  >
    <span
      className="text"
    />
  </a>
</div>
`;
